packages/react/src/components/Table/Table.story.jsx

```diff
@@ -368,7 +396,10 @@ const RowExpansionContent = ({ rowId }) => (
     <ul style={{ lineHeight: '22px' }}>
       {Object.entries(tableData.find((i) => i.id === rowId).values).map(([key, value]) => (
         <li key={`${rowId}-${key}`}>
-          <b>{key}</b>: {value}
+          <b>{key}</b>:{' '}
+          {!React.isValidElement(value) && typeof value === 'object' && value !== null
+            ? JSON.stringify(value, null, 2)
+            : value}
```

Comment:
JSON stringify the object column to prevent React throwing errors about objects being invalid as children when this row is expanded.